test/Graphics/Drawable.test.cpp

```diff
@@ -1,9 +1,49 @@
 #include <SFML/Graphics/Drawable.hpp>
+#include <SFML/Graphics/RenderTexture.hpp>
 
+#include <doctest/doctest.h>
+
+#include <GraphicsUtil.hpp>
 #include <type_traits>
 
 static_assert(!std::is_constructible_v<sf::Drawable>);
 static_assert(!std::is_copy_constructible_v<sf::Drawable>);
 static_assert(std::is_copy_assignable_v<sf::Drawable>);
 static_assert(!std::is_nothrow_move_constructible_v<sf::Drawable>);
 static_assert(std::is_nothrow_move_assignable_v<sf::Drawable>);
+
+class DrawableTest : public sf::Drawable
+{
+public:
+    void draw(sf::RenderTarget&, const sf::RenderStates&) const override
+    {
+        m_called += 1;
+    }
+
+    int callCount() const
+    {
+        return m_called;
+    }
+
+private:
+    mutable int m_called{};
+};
+
+TEST_CASE("[Graphics] sf::Drawable" * doctest::skip(skipDisplayTests))
+{
+    SUBCASE("Construction")
+    {
+        const DrawableTest drawableTest;
+        CHECK(drawableTest.callCount() == 0);
+    }
+
+    SUBCASE("draw()")
+    {
+        const DrawableTest drawableTest;
+        sf::RenderTexture  renderTexture;
+        CHECK(renderTexture.create({32, 32}));
+        CHECK(drawableTest.callCount() == 0);
+        renderTexture.draw(drawableTest);
```

Comment:
Isn't this testing RenderTexture rather than Drawable?

Reply:
Somewhat, but how else do we test `Drawable`? We need something that call `.draw` on the drawable since users can't call that directly.

Reply:
Drawable has no implementation, it is only an interface, so there is no behavior to assert.
However we can assert properties of the interface itself via type_traits such being abstract or having a virtual destructor, or implement some template magic to check that draw is virtual if we want to go that far. (But isn't it only repeating the type definition then?)

Reply:
Can we assert the presence of a protected pure virtual function?

Reply:
> Can we assert the presence of a protected pure virtual function?

I don't know if it can be done. At least deriving the class lets us ensure `draw` is a virtual method by using `override` in the derived class.
Apart from that and static_asserts, we are actually testing the behavior of a test class. Is it useful? What error in Drawable can this detect? :thinking:

Reply:
> Is it useful?

Useful enough to warrant not just closing this PR.

> What error in Drawable can this detect? 🤔

Nothing that won't already be caught my the loads of other classes in the library that also derive from `sf::Drawable`. I know you don't think these tests are worthwhile but do you think this PR is worth closing or are you fine having some relatively low value tests in the library?

Reply:
As we discussed privately, those tests are not so much about catching bugs but rather being a live documentation. Then I understand the value :+1: